### PR TITLE
Try harder to get diagnostics on server launch failure

### LINKED ISSUE #1394: Add an env var or other way to keep temporary server config file
In ServerApplication.scala, if it fails to launch the server it throws an exception which includes the command line used, but the config file gets deleted so you can't just run this command line to debug the problem.

There's a similar issue one level up in sbt remote control (it deletes the launch properties file).

It would greatly aid debugging to have an env var which caused sbt-remote-control to not delete the launch properties and sbt to not delete the server config. Then you can manually run these commands and see why they fail.

Could be as simple as "if (System.getenv("SBT_SAVE_TEMPS") eq null)" in front of the `deleteOnExit` call.

### PR DESCRIPTION
**Pull request by havocp**
Fixes #1394

### CODE REVIEW

**Comment by havocp** [comment]
btw if these changes look ok can you publish an MSERVER-4 with them to see what's wrong on maxime's system?

File: launch/src/main/scala/xsbt/boot/ServerApplication.scala [hunk 1]
```diff
@@ -101,6 +101,7 @@ class StreamDumper(in: java.io.BufferedReader, out: java.io.PrintStream) extends
       case null => ()
       case line =>
         out.println(line)
+        out.flush()
```
Comment (jsuereth) [inline launch/src/main/scala/xsbt/boot/ServerApplication.scala:104]: nice and slow, just the way I like it.

MAYBE we shoudl not use a buffered writer, except it's System.out huh?

File: launch/src/main/scala/xsbt/boot/ServerApplication.scala [hunk 2]
```diff
@@ -132,7 +133,8 @@ object ServerLauncher {
       case None    => throw new RuntimeException("Logic Failure:  Attempting to start a server that isn't configured to be a server.  Please report a bug.")
     }
     val launchConfig = java.io.File.createTempFile("sbtlaunch", "config")
-    launchConfig.deleteOnExit()
+    if (System.getenv("SBT_SERVER_SAVE_TEMPS") eq null)
```
Comment (jsuereth) [inline launch/src/main/scala/xsbt/boot/ServerApplication.scala:136]: Nice hook.

**Comment by jsuereth** [comment]
LGTM, once we get green we can merge.

**Pull request merged**